sys/net/gnrc/link_layer/lorawan/gnrc_lorawan_mcps.c

```diff
@@ -358,7 +373,14 @@ void gnrc_lorawan_event_no_rx(gnrc_lorawan_t *mac)
         return;
     }
 
-    /* Otherwise check if retransmission should be handled */
+    if (mac->mlme.adr && (mac->last_dr != 0)) {
+        mac->mlme.adr_ack = mac->mlme.adr_ack_cnt >= CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT;
+        if (mac->mlme.adr_ack && ((mac->mlme.adr_ack - CONFIG_LORAMAC_DEFAULT_ADR_ACK_LIMIT) \
```

Comment:
The DR should be decrease for the first time when `mac->mlme.adr_ack >= ADR_ACK_LIMIT + ADR_ACK_DELAY`. The snippet still doesn't reflect that.